### ISSUE TITLE
Invalid component name: "pages/product/_slug.vue". Component names should conform to valid custom element name in html5 specification.

### ISSUE THREAD

**Issue by simplenotezy** [nuxt-community/router-module#67]
### Version

[v1.3.0](https://github.com/router-module/releases/tag/v1.3.0)

### Steps to reproduce

I have some dynamic routes. My folder structure is this:

    - pages
     - product
      - _slug.vue

I link to the route like this: 

    <nuxt-link :to="{ name: 'product-slug', params: { slug: product.slug } }">

It works fine, it shows the correct URL and also directs the page fine, however, I am getting an annoying red error in my console:

    [Vue warn]: Invalid component name: "pages/product/_slug.vue". Component names should conform to valid custom element name in html5 specification.

[![enter image description here][1]][1]

I have found this issue, but to little avail: https://github.com/nuxt/nuxt.js/issues/165

  [1]: https://i.stack.imgur.com/EYKRl.png

### What is expected ?

No error should be outputted

### What is actually happening?

An error is being outputted even though it links just fine

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/router-module/issues/c50">#c50</a>)</em></sub></div>

**Comment by ricardogobbosouza** [comment]
Hi @simplenotezy 
I was unable to reproduce your issue

Are you using this module?
Use this module to use router.js instead of pages/ directory

**Comment by sqram** [comment]
FYI, also getting this error on 2.11.0

Started happenign suddenly. I even switched to master branch, and it's happening there too.
Tried deleting  .nuxt, node_modules, and no luck.

 ERROR  [Vue warn]: Invalid component name: "_77180f1e". Component names should conform to valid custom element name in html5 specification.  

Stacktrace shown in image:
https://imgur.com/H2QCSNZ

**Comment by theo-bittencourt** [comment]
Same here. But not on page load. Just happen when click on some link.

**Comment by InerkyJad** [comment]
### The issue, and it's Solution (tested)
actually there is no bug, the only way i found to reproduce the error is to add and export the name from a `page` or a `component`
something like:
```
export default {
  name: "_slug"
}
```
**Note:** this error occurs only when you have a some dynamic routing set up with the page `_slug.vue`, it's important to note that the reason is mostly that your IDE is setting `.vue` files with some initial boilerplate containing the property `name` with the value of the filename, this will result in the name being exported to be `_slug` and we all know that `Nuxt`  uses the names.

the page that you access doesn't have to be `_slug.vue` and exporting `name: '_slug'` to run into the error, just it's existence sometimes (also tested) results in that error, it's strange but i couldn't find the reason for that, the important thing is that you can solve the error just by changing the name to an accepted name (a name that doesn't have `_` or any other none valid characters )

TL;DR
change this:
```
export default {
  name: "_slug"
}
```
to this: 
```
export default {
  name: "someComponentName"
}
```

**Comment by PlusA2M** [comment]
Facing same issue here for `pages/index.vue` and any dynamic route with underscore.

**Comment by arraintxo** [comment]
Hi,

sorry for bumping this thread, but is there any workaround when using `<script setup>` syntax?

**Comment by MartinX3** [comment]
Sadly there is no workaround for the `<script setup>` syntax (with typescript).
(Using nuxtjs 2.16.0 + bridge)

**Comment by arraintxo** [comment]
I already migrated to Nuxt 3:

I had to change path from ```_category/_product.vue``` to ```[category]/[product].vue``` and works perfectly
:)

**Comment by dbssman** [comment]
> Sadly there is no workaround for the `<script setup>` syntax (with typescript). (Using nuxtjs 2.16.0 + bridge)

You can have a second script block, where you the name "trick" like this:

```vue
<template>
</template>

<script>
// extra options like name, layout...
export default {
 name: '<your-page-name-or-whatever>'
}
</script>

<script setup>
// Your normal script setup stuff here
</script>
```

Also remember that the extra script tag has to be in the same lang like your setup tag so if you are using typescript

```vue
<template>
</template>

<script lang="ts">
// extra options like name, layout...
export default {
 name: '<your-page-name-or-whatever>'
}
</script>

<script setup lang="ts">
// Your normal script setup stuff here
</script>
```

Hope that helps :)

**Comment by MartinX3** [comment]
Please read the vue documentation.  
You can write the name information in `setup` as well.

**Comment by dbssman** [comment]
> Please read the vue documentation. You can write the name information in `setup` as well.

Not for me in a nuxt + bridge setup, since defineOptions does not work there
![image](https://github.com/nuxt-community/router-module/assets/82184879/4b3dcacd-97e2-40a9-adc6-e9c9b26c0f92)

While other things like defineProps and defineEmits do work.

So yes, I've read the docu, but afaik nuxt2 with bridge is not fully compatible with all the macros that we have in a vue3^ project :)

**Comment by MartinX3** [comment]
That's sad, especially since a switch to nuxt 3 is not so good since vuetify 3 is still not feature complete.  

I migrated away from nuxt to quasar because they have the same features and migrate to newer vuejs versions instantly with their framework and gui components.
Also they have less boilerplate code.

**Comment by dbssman** [comment]
> That's sad, especially since a switch to nuxt 3 is not so good since vuetify 3 is still not feature complete.
> 
> I migrated away from nuxt to quasar because they have the same features and migrate to newer vuejs versions instantly with their framework and gui components. Also they have less boilerplate code.

Yes, we also have the pain of having vuetify and no good way of migrating, but we are taking the chance to just move completely out of a material design framework and build our own components using tailwind (so that we keep it framework and framework version agnostic).

In general, a lot of things are improvable while migrating to nuxt3 but we have to live with what is available.

if someone wants to know, general things that do not work:
```vue
<script setup lang="ts">
//Typescript
import type { XInterface } from 'y'

//this does not work, it is not able to infer your props properly in nuxt + bridge
interface Props extends XInterface {
//more things
}

defineProps<Props>()

//something like this does work
import type {XInterface} from 'y'

interface Props {
attribute:XInterface
//moreprops
}

//Macros
defineOptions
defineSlots
defineModel (easily replaceable with a writable computed)
</script>
```

Generic components do not work either, and in general trying to work with imported types for props is kinda annoying.

In any case, this is not a part of the router-module nor this thread, But I hope it helps someone
